File: backend/src/main/java/com/incubyte/salarymanagement/service/AnalyticsService.java
package com.incubyte.salarymanagement.service;

import com.incubyte.salarymanagement.dto.CountrySalaryStat;
import com.incubyte.salarymanagement.dto.DepartmentSalaryStat;
import com.incubyte.salarymanagement.dto.HeadcountSummary;
import com.incubyte.salarymanagement.dto.PayrollByCurrency;
import com.incubyte.salarymanagement.dto.SalaryBandStat;

import java.util.List;

public interface AnalyticsService {

    List<DepartmentSalaryStat> averageSalaryByDepartment();

    List<CountrySalaryStat> averageSalaryByCountry();

    List<SalaryBandStat> salaryBandDistribution();

    HeadcountSummary headcountSummary();

    List<PayrollByCurrency> payrollByCurrency();
}

File: frontend/src/app/core/services/analytics.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { CountrySalaryStat, DepartmentSalaryStat, SalaryBandStat } from '../models/analytics.model';

@Injectable({ providedIn: 'root' })
export class AnalyticsService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = '/api/v1/analytics';

  averageSalaryByDepartment(): Observable<DepartmentSalaryStat[]> {
    return this.http.get<DepartmentSalaryStat[]>(`${this.baseUrl}/avg-salary-by-department`);
  }

  averageSalaryByCountry(): Observable<CountrySalaryStat[]> {
    return this.http.get<CountrySalaryStat[]>(`${this.baseUrl}/avg-salary-by-country`);
  }

  salaryBandDistribution(): Observable<SalaryBandStat[]> {
    return this.http.get<SalaryBandStat[]>(`${this.baseUrl}/salary-bands`);
  }
}
